app/models/service_ansible_playbook.rb

```diff
@@ -68,12 +73,16 @@ def on_error(action)
 
   private
 
+  def retain_resources_on_retirement?
+    %w[no_without_playbook no_with_playbook].include?(options.fetch_path(:config_info, :retirement, :remove_resources))
```

Comment:
Realized I already commented on this line, but looking at the options it seems like it is just:
`options.fetch_path(:config_info, :retirement, :remove_resources).to_s.starts_with("no_")`

Not a required change, but seems simpler and maybe more future proof.